crates/service-protocol/src/discovery.rs

```diff
@@ -341,11 +361,24 @@ impl ServiceDiscovery {
                 Err(e) => e,
             };
 
+            if is_possible_h1_only_error(&e) {
+                if let Some(next_retry_interval) = retry_iter.next() {
+                    warn!(
+                        "Detected possible HTTP1.1 endpoint when discovering deployment at address '{}'. Retrying with HTTP1.1 in {} seconds",
+                        address,
+                        next_retry_interval.as_secs(),
+                    );
+                    try_http1 = true;
+                    tokio::time::sleep(next_retry_interval).await;
```

Comment:
In case of downgrading to `http1.1`, should we retry right away w/o the sleep?

Reply:
Ok, I see there is a potential problem if there is another error that we wrongly detect as a possible h1 error. In this case and w/o the sleep, we would not pause between retries.

Reply:
The first retry is instant or near instant anyway afaict, so it seemed a safer bet